### PR TITLE
Update Katib manifests with the latest changes

### PR DESCRIPTION
**Pull request by andreyvelich**
**Description of your changes:**
This PR updates Katib manifests with the latest changes and also changes some parameters according to Katib repository to be consistent.

@richardsliu Do we need to update port to 8080 [here](https://github.com/kubeflow/manifests/blob/master/katib/katib-controller/overlays/istio/katib-ui-virtual-service.yaml#L20), since we are using `targetPort: 8080` for Katib UI or we don't need to make this change?

Please, take a look.
/assign @richardsliu @johnugeorge 
/cc @hougangliu @gaocegege 

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/853)

### CODE REVIEW

File: katib/katib-controller/base/katib-controller-rbac.yaml [hunk 1]
```diff
@@ -54,13 +54,10 @@ rules:
   resources:
   - experiments
   - experiments/status
-  - experiments/finalizers
```
Comment (richardsliu) [inline katib/katib-controller/base/katib-controller-rbac.yaml]: Why are these removed?

File: katib/katib-controller/base/katib-mysql-pv.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (richardsliu) [inline katib/katib-controller/base/katib-mysql-pv.yaml]: Is this needed?

Comment (johnugeorge) [reply katib/katib-controller/base/katib-mysql-pv.yaml]: PV and pvc changes are not required

Comment (andreyvelich) [reply katib/katib-controller/base/katib-mysql-pv.yaml]: I think it is useful when you are not using GCP and you don't have StorageClass on your cluster. We have this PV in Katib manifest.
What do you think @richardsliu @johnugeorge ?

File: katib/katib-controller/base/katib-controller-rbac.yaml (hunk 1 above)
Comment (andreyvelich) [reply katib/katib-controller/base/katib-controller-rbac.yaml]: We don't have these roles in controller rbac: https://github.com/kubeflow/katib/blob/master/manifests/v1alpha3/katib-controller/rbac.yaml#L54. Do we need it?

File: katib/katib-controller/base/katib-mysql-pv.yaml (hunk 2 above)
Comment (johnugeorge) [reply katib/katib-controller/base/katib-mysql-pv.yaml]: But then, need not be in base in that case.

Comment (andreyvelich) [reply katib/katib-controller/base/katib-mysql-pv.yaml]: @johnugeorge @richardsliu Ok, I removed it from base.
